### DISCUSSION TITLE
Direction-based Vision

### DISCUSSION THREAD

**Discussion by cosmiccoincidence** [RE-SS3D/SS3D#1159] [Game Design]
From here: https://trello.com/c/O6q1yahq/125-fov

**Vote on this topic if you like the idea or not.**

## Summary

![Directional Vision](https://user-images.githubusercontent.com/41941008/229215272-26ae7336-5394-4fe8-aaeb-cb9166c11044.png)

Vision is blurred to your sides and also darkened fully behind you with the addition of not rendering items and entities. [See image] Maybe reduce the radius of the Blurred (purple line) layer.

## Details

When wearing masks or large helmets the angle of the FOV could be reduced, and things like welding masks would be reduced even more and maybe the radius as well.

Different species may have different default FOV angles (moths might have 300 degree vision).

Mechs may have proximity cameras that allow the pilot to see full 360 degrees, these could get damaged in a fight, also be upgraded with various sensor types. Exosuits like the ripley would be less sophisticated and have large truck-like side mirrors which could be broken off in a fight (if the left side gets broken off they lose vision of their back left corner).

Angle/radius of FOV could also be reduced during concussions or states of half-blindness. Consider various levels of blindness. If we end up separating the players eyes to behave individually, we could have various states of blindness for both eyes (left or right side of screen).

Ghosts would have no FOV angle or color restrictions.

## Extras

This could allow for things to play off the fact that people's vision is limited. You could now try to sneak through a room when a player's back is turned. Standing within 2 tiles of a mirror would allow seeing behind you (or maybe even when holding up a handheld mirror).

New items like [360 syndicate spy glasses](https://edc-packs.com/wp-content/uploads/2017/10/59dc6665793a0-0.jpg), or a back-stabbing syndicate dagger that deals 3X damage to a player's back but .5X damage anywhere else.

Even creatures/antagonists that react based on being seen or not. Like being frozen in place for a short period or becoming violent after being seen.

**There are many ways this could impact gameplay. If you have any thoughts or ideas feel free to share.**

**Comment by CapoFantasma97** [comment]
It's very hard to tell how good or bad of an idea it could be without actually experiencing it in a test run.

For the specific example provided, maybe it could be improved by also giving a very small area around the player to give clear vision, so that you can see what's immediately under or behind you for quality of life purposes (I'm sure any backstabbers won't be affected by you seeing them in the last 100 milliseconds before the disaster).

![immagine](https://github.com/RE-SS3D/SS3D/assets/9368910/01bf5771-4c73-46b0-9e7f-c0835de2be6f)
